replication.go

```diff
@@ -402,12 +402,15 @@ func (r *Raft) heartbeat(s *followerReplication, stopCh chan struct{}) {
 
 		start := time.Now()
 		if err := r.trans.AppendEntries(peer.ID, peer.Address, &req, &resp); err != nil {
-			r.logger.Error("failed to heartbeat to", "peer", peer.Address, "error", err)
+			nextBackoffTime := cappedExponentialBackoff(failureWait, s.failures, maxFailureScale, r.config().HeartbeatTimeout/2)
```

Comment:
The code changed from using the local `failures` variable to `s.failures`. What's the difference here?

Reply:
Hm, I'm not sure -- it was an unintentional change, so I'll revert it. It's interesting though that s.failures is used for all the other backoffs, but not for the heartbeat backoffs. `s` is followerReplication, and the comment above the struct says: 

`// followerReplication is in charge of sending snapshots and log entries from
// this leader during this particular term to a remote follower.`

and since heartbeats are empty log entries, it makes sense that s.failures would get incremented and used for backoff on failed heartbeats. On the other hand, s.failures >= failures because failures is local in `heartbeat` and initialized at 0, so it's probably better to revert to failures. 

Thanks for catching!